### PR TITLE
[WORKFLOWS-560] add rnaseq human pipeline for workflows-560

### PR DESCRIPTION
**Pull request by danlu1**
Problem:
The Sage-Bionetworks-Workflows/nf-synstage is going to be deprecated.

Solution:
Replace Sage-Bionetworks-Workflows/nf-synstage with Sage-Bionetworks-Workflows/nf-synapse and use entry_name="NF_SYNSTAGE". Also, to use the entry point, remove the restriction on nextflow version for staging step to allow using the most current version of nextflow tower.

### CODE REVIEW

**Comment by BryanFauble** [comment]
Please add a description when you get a chance @danlu1

File: local/ntap/workflows-560-rnaseq-human.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            run_number=1,
+            output_number=1,
+        )
+    ]
+
+def stage_samplesheet(syn: Synapse, dataset: Dataset):
+    """Download the samplesheet from synapse and upload it to S3 in the location where synstage
+    is going to grab the file.
+
+    :param syn: The logged in synapse instance.
```
Comment (BryanFauble) [inline local/ntap/workflows-560-rnaseq-human.py]: Return type of None. And could you change these over to google style docstrings. That'll kick off the overall conversion to it for this repo.

File: local/ntap/workflows-560-rnaseq-human.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            "fasta": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/WholeGenomeFasta/genome.fa",
+            "gtf": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.gtf",
+            "gene_bed": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.bed",
+            "star_index": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/STARIndex/",
+            # TODO: These could not be found in `s3://sage-igenomes/igenomes/Homo_sapiens/`
+            # "transcript_fasta": "",
+            # "salmon_index": "",
+            # "rsem_index": "",
+        },
+        pre_run_script="NXF_VER=23.10.0",
```
Comment (BryanFauble) [inline local/ntap/workflows-560-rnaseq-human.py:180]: @BWMac If I recall what you had said, we shouldn't we specifying this manually?

File: local/ntap/workflows-560-rnaseq-human.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        nextflow_config=Path("local/ntap/nextflow-rnaseq.config").read_text(),
+    )
+
+
+def prepare_synindex_launch_info(dataset: Dataset) -> LaunchInfo:
+    """Generate LaunchInfo for nf-synindex workflow run."""
+    return LaunchInfo(
+        run_name=dataset.synindex_run_name,
+        pipeline="Sage-Bionetworks-Workflows/nf-synindex",
+        revision="disable_wave",
```
Comment (BryanFauble) [inline local/ntap/workflows-560-rnaseq-human.py:197]: Is synindex available in
pipeline="Sage-Bionetworks-Workflows/nf-synapse"
?

Comment (danlu1) [reply local/ntap/workflows-560-rnaseq-human.py:197]: Not yet. It's Brad's goal to add synindex to nf-synapse in this sprint.

File: local/ntap/workflows-560-rnaseq-human.py (hunk 2 above)
Comment (danlu1) [reply local/ntap/workflows-560-rnaseq-human.py:180]: We talked about the nextflow version for `synstage` last meeting. I confirmed with Brad that we can keep using the specified version for other steps for this round. But we can open a new ticket to update this part later.

File: local/ntap/workflows-560-rnaseq-human.py (hunk 1 above)
Comment (danlu1) [reply local/ntap/workflows-560-rnaseq-human.py]: Thanks for catching it. I updated it in the latest commit.

**Review by BryanFauble — APPROVED** [review]
Changes LGTM!

**Pull request merged**